pettingzoo/utils/env.py

```diff
@@ -354,11 +351,10 @@ def observation_space(self, agent: AgentID) -> gym.Space:
         warnings.warn(
             "Your environment should override the observation_space function. Attempting to use the observation_spaces dict attribute."
         )
-        return self.observation_spaces[agent]
+        return self.observation_spaces[agent]  # type: ignore
```

Comment:
Why the ignores here? Can you add some comments around all ignores to explain them?

Reply:
I was able to remove almost all `# type: ignore` statements including this one. There are still 2 I wasn't able to remove, I'll make a review comment to keep them in mind.